qualtran/resource_counting/classify_bloqs.py

```diff
@@ -0,0 +1,104 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import importlib
+import inspect
+import sys
+from collections import defaultdict
+from typing import Dict, Optional, Tuple, Union
+
+import sympy
+
+from qualtran import Adjoint, Bloq
+from qualtran.bloqs.basic_gates import CSwap, TGate, Toffoli
+from qualtran.bloqs.reflection import Reflection
+from qualtran.resource_counting.generalizers import (
+    ignore_alloc_free,
+    ignore_cliffords,
+    ignore_partition,
+    ignore_split_join,
+)
+from qualtran.resource_counting.t_counts_from_sigma import (
+    _get_all_rotation_types,
+    t_counts_from_sigma,
+)
+
+
+def _get_all_bloqs_in_module(bloq_module: str) -> Tuple[Bloq]:
+    """Returns all classes defined in bloqs `bloq_module`."""
+    importlib.import_module(bloq_module)
+    all_members = [v for (_, v) in inspect.getmembers(sys.modules[bloq_module], inspect.isclass)]
```

Comment:
what if someone defines a class that isn't a subclass of `Bloq`?

Reply:
did you test that this isn't mis-organizing bloqs imported by other bloqs? Take a look at dev_tools/bloq_finder.py

Reply:
When would this happen? The goal is to classify bloqs, are any bloqs not subclasses of Bloq?

Reply:
there might be members of `sys.modules[bloq_module]` that are classes but not bloqs that will be returned

Reply:
On your second point, that's a good test. But it works:

``` python
hwp =  _get_all_bloqs_in_module('qualtran.bloqs.rotations.hamming_weight_phasing')
```
returns 

```
 (<class 'qualtran.bloqs.rotations.hamming_weight_phasing.HammingWeightPhasing'>, <class 'qualtran.bloqs.rotations.hamming_weight_phasing.HammingWeightPhasingViaPhaseGradient'>)
```
(i.e. no ZPowGate or any other basic gates used in that module)

The following line checks that `bloq_module` is in the list of members which prunes anything out which doesn't have 'qualtran.bloqs.hamming_weight_phasing` in it.

Reply:
> there might be members of `sys.modules[bloq_module]` that are classes but not bloqs that will be returned

Ah, I can definitely add a check that it is a bloq.